### ISSUE TITLE
Research request (enterprise): third-party (not Identity) JWT based redirects no longer working

### ISSUE THREAD

**Issue by overlordofmu** [netlify/netlify-redirect-parser#439]
**Description**

An Enterprise customer has had a working site access control system to only allow certain IP addresses to connect since March 11th. It was working and then suddenly stopped working without any changes or new deploys.

There was a new deploy on 2022-05-03 to allow my IP address of `98.232.139.254` to the function that sets the JWT cookie. However, this JWT site filtering stopped working before that deploy.

The list of IP addresses is configured as an environment variable:

```text
[64] pry(main)> Site.find_by_subdomain("ts-education").repo.env["ALLOWED_IPS"]
"193.138.43.53,77.234.159.44,213.229.248.10,193.77.253.105,77.234.137.105,193.95.247.82,89.143.12.250,193.189.160.24,98.232.139.254"
```

Here is an example `x-nf-request-id`: 01G271ZWP6ADH1XRE8N83EXFVN

Here it is in Humio:
 
https://cloud.us.humio.com/netlify-us-production/search?query=%24ut_raw%28%29%20%7C%2001G271ZWP6ADH1XRE8N83EXFVN&live=false&start=1651651860000&end=1651651920000

This shows the site redirecting back to the function, even though the JWT cookie was sent. The cookie sent by my browser was:

```text
cookie: nf_jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlY2ZlYjlkLTkyYWYtNGFhOS05NzI0LTFlZTMxNzkyNjA2ZSIsImV4cCI6MTY1MTY1NTUwMCwiZW1haWwiOiJuby1yZXBseUBwLW0uc2kiLCJhcHBfbWV0YWRhdGEiOnsicHJvdmlkZXIiOiJpcGNoZWNrIiwiYXV0aG9yaXphdGlvbiI6eyJyb2xlcyI6WyJhdXRob3JpemVkIl19fSwidXNlcl9tZXRhZGF0YSI6e30sImlhdCI6MTY1MTY1MTkwMH0.Zb7Gui5SgdNfletVsna_KPVfmDIsrkRoYaGPxY2A2O4
```

Using an online JWT decoder, I see this:

```text
{
  "id": "5ecfeb9d-92af-4aa9-9724-1ee31792606e",
  "exp": 1651655500,
  "email": "no-reply@p-m.si",
  "app_metadata": {
    "provider": "ipcheck",
    "authorization": {
      "roles": [
        "authorized"
      ]
    }
  },
  "user_metadata": {},
  "iat": 1651651900
}
```

That matches the format in our documentation here:

https://docs.netlify.com/visitor-access/role-based-access-control/#external-providers

Quoting those docs:

```text
{
  "id": "some id",
  "exp": 1602522810,
  "app_metadata": {
    "authorization": {
      "roles": ["admin","editor"]
    }
  }
}

```

The redirect rules are defined like so:

```text
/* 200! Role=authorized
/* /.netlify/functions/anon 401!
```

They are deployed like this:

```text
[66] pry(main)> Site.find_by_subdomain("ts-education").redirects
[
    [0] {
          :path => "/.netlify/git/*",
            :to => "https://git-gateway.services.netlify.com/:splat",
         :proxy => true,
         :force => true,
        :status => 200,
          :sign => {
                    :type => "jws",
                  :secret => "NnRYpxdsBT6L6vL59zjgBK9fR",
              :jwt_secret => "eb37c4402f9f6900cec83dc5d8d6322c91d31bd2f572aac30707b063bddd9a7549e0ce7dd36ac069ff0e8894aa9ef07a833a",
            :extra_fields => {
                    :netlify_id => "53acada2-b60a-4778-b13d-b893aeed79cb",
                            :id => "1e5ae983-7146-4c13-88f4-416563baba99",
                      :site_url => "https://program-izobrazevanj.ts.telekom.si",
                :deploy_context => "production",
                           :env => "stable"
            }
        }
    },
    [1] {
              "path" => "/*",
                "to" => "/:splat",
            "status" => 200,
             "force" => true,
        "conditions" => {
            "Role" => "authorized"
        },
           "pattern" => "^/?(.*)/?$",
          "captures" => [
            [0] ":splat"
        ]
    },
    [2] {
            "path" => "/*",
              "to" => "/.netlify/functions/anon",
          "status" => 401,
           "force" => true,
         "pattern" => "^/?(.*)/?$",
        "captures" => [
            [0] ":splat"
        ]
    }
]
```

To summarize, this was working and stopped with no changes on their side.

The current behavior is that I'm redirected back to the function even with the valid cookie. This results in a loop of endless 302s as the function sets the cookie but then the site redirect back to the function like I am not authorized.

**Questions we hope to answer**

Why did the redirect stop working?
What must be done to get the redirect working again?

**Additional information**

I've attached a HAR file of the issue occurring below:

[ts-education-2022-05-04-from-luke-at-netlify.zip](https://github.com/netlify/netlify-redirect-parser/files/8619055/ts-education-2022-05-04-from-luke-at-netlify.zip)

**Comment by overlordofmu** [comment]
**Link to ticket**

https://netlify.zendesk.com/agent/tickets/92414

**Comment by fool** [comment]
@mrdg to take a look to see if some higher-level change might have impacted more people around 3 May with JWT's on the **regular CDN** where the customer must be testing this site (since no DNS is configured for custom domain).  The "answer the identity question, if nothing has changed" is not in immediate scope (later this week, perhaps;  not highest-priority today).

**Comment by mrdg** [comment]
I looked at the site (`53acada2-b60a-4778-b13d-b893aeed79cb`) in IRB and it doesn't have a JWT secret set:

```
[3] pry(main)> s.encrypted_jwt_secret
nil
[4] pry(main)> s.encrypted_account_jwt_secret
nil
```

I think is required for RBAC to work? Could we ask them or check if they had that configured before?

**Comment by overlordofmu** [comment]
We have a root cause! They downgraded the team from Business to Pro and Pro doesn't include third-party JWT RBAC. You can use Netlify Identity but not third-party JWTs on Pro.

To summarize, when they downgraded the third-party JWTs stopped working.

We might want to consider showing a warning about the things that will be disabled before we allow people to proceed. Thoughts on that? If so, I'll make a feature request (or bug) if one doesn't already exist.
